device_local.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+// +build local
+
+package main
+
+import (
+	"io"
+
+	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
+)
+
+type deviceConfig struct {
+	rootfsPartA string
+	rootfsPartB string
+}
+
+type device struct {
+	BootEnvReadWriter
+	Commander
+	parts partitions
+}
+
+func NewDevice(env BootEnvReadWriter, sc StatCommander, config deviceConfig) *device {
+	partitions := partitions{
+		StatCommander:     sc,
+		BootEnvReadWriter: env,
+		rootfsPartA:       config.rootfsPartA,
+		rootfsPartB:       config.rootfsPartB,
+		active:            "",
+		inactive:          "",
+	}
+	device := device{env, sc, partitions}
+	return &device
+}
+
+func (d *device) Reboot() error {
+	log.Infof("reboot")
+	return nil
+}
+
+func (d *device) Rollback() error {
```

Comment:
My point is that you have dedicated `_local` versions of `Rollback`, `InstallUpdate`, `EnableUpdatedPartition`, `CommitUpdate` and `HasUpdate`, all of which seem to be unneeded if you point the `RootfsPartA` and `RootfsPartB` to local files using the configuration file. And assuming that `BootEnv` is behaving as expected when using the `_local` version.